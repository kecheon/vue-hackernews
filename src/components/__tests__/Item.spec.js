import { mount } from '@vue/test-utils'
import Item from '../Item.vue'

describe('Item.vue', () => {
  test('renders item.score', () => {
    const item = {
      score: 10
    }
    const wrapper = mount(Item, {
      propsData: { item }
    })
    expect(wrapper.text()).toContain(item.score)
  })

  test('renders item.by', () => {
    const item = {
      by: 'some author'
    }
    const wrapper = mount(Item, {
      propsData: { item }
    })
    expect(wrapper.text()).toContain(item.by)
  })

  test('renders item.url', () => {
    const item = {
      url: 'some title'
    }
    const wrapper = mount(Item, {
      propsData: { item }
    })
    expect(wrapper.text()).toContain(item.url)
  })

  test('renders an a tag containing item.title', () => {
    const item = {
      title: 'some title',
      url: 'http://some-url.com'
    }
    const wrapper = mount(Item, {
      propsData: { item }
    })
    expect(wrapper.find('a').text()).toEqual(item.title)
  })

  test('renders an a tag with href item.url', () => {
    const item = {
      url: 'http://some-url.com'
    }
    const wrapper = mount(Item, {
      propsData: { item }
    })
    const aWrapper = wrapper.find('a')
    expect(aWrapper.element.getAttribute('href')).toBe(item.url)
  })

  test('renders the host name', () => {
    const item = {
      url: 'https://some-url.com/with-paths'
    }
    const wrapper = mount(Item, {
      propsData: {
        item
      }
    })
    expect(wrapper.text()).toContain('(some-url.com)')
  })

  test('renders the time since the last post', () => {
    const dateNow = jest.spyOn(Date, 'now')
    const dateNowTime = new Date('2018')

    dateNow.mockImplementation(() => dateNowTime)

    const item = {
      time: (dateNowTime / 1000) - 600
    }
    const wrapper = mount(Item, {
      propsData: {
        item
      }
    })
    dateNow.mockRestore()
    expect(wrapper.find('.time').text()).toBe('10 minutes ago')
  })

  test('renders correctly', () => {
    const dateNow = jest.spyOn(Date, 'now')
    const dateNowTime = new Date('2018')

    dateNow.mockImplementation(() => dateNowTime)

    const item = {
      by: 'eddyerburgh',
      id: 11122233,
      score: 10,
      time: (dateNowTime / 1000) - 600,
      title: 'vue-test-utils is released',
      type: 'story',
      url: 'https://vue-test-utils.vuejs.org/'
    }
    const wrapper = mount(Item, {
      propsData: {
        item
      }
    })
    dateNow.mockRestore()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly when item has no url and has type top', () => {
    const dateNow = jest.spyOn(Date, 'now')
    const dateNowTime = new Date('2018')

    dateNow.mockImplementation(() => dateNowTime)

    const item = {
      by: 'eddyerburgh',
      id: 11122233,
      score: 10,
      time: (dateNowTime / 1000) - 600,
      title: 'vue-test-utils is released',
      type: 'top'
    }
    const wrapper = mount(Item, {
      propsData: {
        item
      }
    })
    dateNow.mockRestore()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
