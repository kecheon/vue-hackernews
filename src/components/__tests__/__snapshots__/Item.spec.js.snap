// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item.vue renders correctly 1`] = `
"<li class=\\"news-item\\"><span class=\\"score\\">10</span> <span class=\\"title\\"><a href=\\"https://vue-test-utils.vuejs.org/\\" target=\\"_blank\\" rel=\\"noopener\\">vue-test-utils is released</a> <span class=\\"host\\"> (vue-test-utils.vuejs.org)</span></span> <br> <span class=\\"meta\\"><span class=\\"by\\">
      by <router-link to=\\"/user/eddyerburgh\\">eddyerburgh</router-link></span> <span class=\\"comments-link\\">
      | <router-link to=\\"/item/11122233\\"> comments</router-link></span> <span class=\\"time\\">
    10 minutes ago
  </span></span> <!----></li>"
`;

exports[`Item.vue renders correctly when item has no url and has type top 1`] = `
"<li class=\\"news-item\\"><span class=\\"score\\">10</span> <span class=\\"title\\"><router-link to=\\"/item/11122233\\">vue-test-utils is released</router-link></span> <br> <span class=\\"meta\\"><span class=\\"by\\">
      by <router-link to=\\"/user/eddyerburgh\\">eddyerburgh</router-link></span> <span class=\\"comments-link\\">
      | <router-link to=\\"/item/11122233\\"> comments</router-link></span> <span class=\\"time\\">
    10 minutes ago
  </span></span> <span class=\\"label\\">top</span></li>"
`;
